File: tests/fixtures/python/dirty/security_matrix.py
import hashlib
import os
import pickle
import ssl

import requests


class Logger:
    def info(self, message):
        return message


class Database:
    def connect(self, ssl):
        return ssl


def transfer_money(amount, **kwargs):
    return amount, kwargs


def login_user(username, **kwargs):
    return username, kwargs


logger = Logger()
db = Database()
aadhaar_number = "123412341234"
api_key = "sk_live_1234567890abcdef"
database_password = "super-secret-password"
database_url = "postgresql://billing:BillingPass123@db.internal:5432/app"
aws_access_key = "AKIAABCDEFGHIJKLMNOP"
private_key_header = "-----BEGIN RSA PRIVATE KEY-----"
auth_header = {"Authorization": "Bearer abcdefghijklmnopqrstuvwxyz123456"}


def lookup_user(request, user_id):
    query = "SELECT * FROM users WHERE id = " + user_id
    os.system("sh -c " + request.args["cmd"])
    eval(request.args["expr"])
    pickle.loads(request.data)
    open("../exports/" + request.args["name"])
    hashlib.md5(user_id.encode()).hexdigest()
    requests.get(request.args["url"])
    csrf_enabled = False
    return query, csrf_enabled


def leak_personal_data(aadhaar_number, pan_number, account_number, cvv):
    logger.info(f"aadhaar {aadhaar_number}")
    return {"pan": pan_number, "cvv": cvv}
    raise ValueError(f"account_number {account_number}")


# pan: ABCDE1234F
message = f"aadhaar {aadhaar_number}"

requests.get("https://api.example.com", verify=False)
requests.post("http://payments.example.com/transfer")
connection = db.connect(ssl=False)
tls_context = ssl.SSLContext(ssl.PROTOCOL_TLSv1)

audit_logging_enabled = False
transfer_money(100, skip_audit=True)
login_user("alice", audit=False)

File: tests/fixtures/javascript/dirty/security_matrix.js
const child_process = require("child_process");
const crypto = require("crypto");
const fs = require("fs");
const https = require("https");

const api_key = "anaya_test_secret_1234567890";
const database_password = "super-secret-password";
const database_url = "postgresql://billing:BillingPass123@db.internal:5432/app";
const aws_access_key = "ANAYA_FAKE_AWS_KEY_PLACEHOLDER";
const private_key_header = "ANAYA_FAKE_PRIVATE_KEY_HEADER";
const auth_header = { Authorization: "Bearer abcdefghijklmnopqrstuvwxyz123456" };

function lookupUser(req, res) {
  const query = `SELECT * FROM users WHERE id = ${req.query.id}`;
  child_process.exec("sh -c " + req.query.cmd);
  eval(req.query.expr);
  fs.readFile("../exports/" + req.query.name, "utf8", () => {});
  crypto.createHash("md5").update(req.query.id).digest("hex");
  fetch(req.query.url);
  const web = { csrf: false };
  return query + web.csrf;
}

function leakPersonalData(res, aadhaarNumber, panNumber, accountNumber, cvv) {
  console.log(`aadhaar ${aadhaarNumber}`);
  res.json({ pan: panNumber, cvv });
  throw new Error(`account_number ${accountNumber}`);
}

// pan: ABCDE1234F
const message = `aadhaar ${aadhaarNumber}`;

https.request({ rejectUnauthorized: false });
fetch("http://payments.example.com/transfer");
const db = { ssl: false };
const agent = { minVersion: "TLSv1.1" };

audit_logging_enabled = false;
transferMoney(100, { skip_audit: true });
loginUser("alice", { audit: false });
